packages/zoe/src/evalContractCode.js

```diff
@@ -1,77 +1,74 @@
-/* global replaceGlobalMeter, registerEndOfCrank */
-import { evaluateProgram } from '@agoric/evaluate';
 import Nat from '@agoric/nat';
 import harden from '@agoric/harden';
 
-import { makeMeter } from '@agoric/transform-metering/src/meter';
 import produceIssuer from '@agoric/ertp';
-import { assert, details } from '@agoric/assert';
+import { assert } from '@agoric/assert';
 import { producePromise } from '@agoric/produce-promise';
 import { sameStructure } from '@agoric/same-structure';
+import { importBundle } from '@agoric/import-bundle';
+import { HandledPromise } from '@agoric/eventual-send';
 
-const evaluateStringToFn = (functionSrcString, endowments) => {
-  assert.typeof(functionSrcString, 'string');
-  const nestedEvaluate = src =>
-    evaluateProgram(src, { ...endowments, nestedEvaluate });
-  const fn = nestedEvaluate(`(${functionSrcString})`);
-  assert.typeof(
-    fn,
-    'function',
-    details`"${functionSrcString}" must be a string for a function, but produced ${typeof fn}`,
-  );
-  return fn;
-};
-
-const evalContractCode = (code, additionalEndowments) => {
+const evalContractBundle = (bundle, additionalEndowments, vatPowers) => {
   // Make the console more verbose.
   const louderConsole = {
     ...console,
     log: console.info,
   };
+  function myRequire(what) {
+    if (what === '@agoric/harden') {
+      return harden;
+    }
+    throw Error(`require(${what}) not implemented`);
+  }
+  harden(myRequire);
+
+  // TODO: this should really only be console and HandledPromise
   const defaultEndowments = {
+    assert,
     console: louderConsole,
     harden,
+    Nat,
+    produceIssuer,
     producePromise,
-    assert,
     sameStructure,
-    produceIssuer,
-    Nat,
+    HandledPromise,
```

Comment:
It should be straightforward to convert the rest to imports. Wanna create an issue and assign step one to me?

Reply:
done, thanks